Bullwinkle.class.nut

```diff
@@ -162,11 +166,20 @@ class Bullwinkle {
         if (_watchdogTimer == null) _watchdog();
 
         // Increment the tries
-        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND) {
-        	message.tries++;
+        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND && message.id in _packages) {
+		        _packages[message.id]._tries++;
+        }
+
+        if(imp.getmemoryfree() > _settings.lowMemoryThreshold && (_isAgent() || server.isconnected())){
```

Comment:
It was left rather open ended in Issue #18.  This is essentially a 2 line implementation that adds very little library bloat, so I went ahead and snuck it in.